scipy/optimize/_numdiff.py

```diff
@@ -516,7 +516,19 @@ def _dense_difference(fun, x0, f0, h, use_one_sided, method):
         else:
             raise RuntimeError("Never be here.")
 
-        J_transposed[i] = df / dx
+        if np.count_nonzero(dx == 0.0):
+            # this can happen for entries with lb == ub
+            with np.errstate(invalid='ignore'):
```

Comment:
Does this substantially slow things down? If not, I'd consider not branching. It looks like this code would work even if all of `dx` is nonzero, right?

Reply:
Or it looks like you might have already considered that.

Reply:
I wasn't sure how much overhead was taken by setting the errstate. I thought it was best to avoid doing that unless we really had to

Reply:
```
In [3]: def f():
   ...:     return 1 + 1
   ...:

In [4]: def g():
   ...:     with np.errstate(invalid='ignore'):
   ...:         return 1+1

In [6]: timeit f()
171 ns ± 10.4 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [7]: timeit g()
12.6 µs ± 651 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```
So I guess there's some. If this is only once per Jacobian evaluation, there might be bigger fish to fry, but it does justify the branching.